plugins/text-editor-resources/src/components/extension/inlineComment.ts

```diff
@@ -0,0 +1,1004 @@
+  }
+})
+
+interface InlineCommentDecoratorState {
+  commentThreads: Map<string, Thread>
+  commentViews: Map<string, InlineCommentView>
+
+  decorationSet: DecorationSet
+  component?: AnySvelteComponent
+  displayMode: InlineCommentDisplayMode
+  displayModeConstraint: InlineCommentDisplayMode
+
+  pointer: PointerState
+
+  pendingComment?: { thread: Thread, selection: Selection }
+}
+
+const inlineCommentDecoratorKey = new PluginKey('inline-comment-decorator')
+
+function getCommentDecoratorState (editorState: EditorState): InlineCommentDecoratorState | undefined {
+  return inlineCommentDecoratorKey.getState(editorState)
+}
+
+export function InlineCommentDecorator (options: InlineCommentExtensionOptions): Plugin {
+  return new Plugin<InlineCommentDecoratorState>({
+    key: inlineCommentDecoratorKey,
+    props: {
+      decorations (state) {
+        return this.getState(state)?.decorationSet
+      },
+      handleDOMEvents: {
+        mousemove: handleInlineCommentMouseHover
+      },
+      transformPasted: (slice) => {
+        const nodes: Node[] = []
+        slice.content.forEach((node) => {
+          nodes.push(removeMarkFromNode(node, 'inline-comment'))
+        })
+        return new Slice(Fragment.fromArray(nodes), slice.openStart, slice.openEnd)
+      }
+    },
+    state: {
+      init (config, state) {
+        const decoratorState: InlineCommentDecoratorState = {
+          commentThreads: new Map(),
+          decorationSet: DecorationSet.empty,
+          displayMode: 'compact',
+          displayModeConstraint: 'compact',
+          commentViews: new Map(),
+          pointer: {
+            hover: new Set(),
+            focus: new Set()
+          }
+        }
+        return buildCommentDecoratorState(options, decoratorState, state, state)
+      },
+      apply (tr, prev, oldState, newState) {
+        return buildCommentDecoratorState(options, prev, oldState, newState, tr)
+      }
+    },
+    view (view) {
+      return initCommentDecoratorView(options, view)
+    }
+  })
+}
+
+function initCommentDecoratorView (options: InlineCommentExtensionOptions, view: EditorView): PluginView {
+  const destructors = [
+    () => {
+      getCommentDecoratorState(view.state)?.commentViews?.forEach((view) => {
+        view.destroy()
+      })
+    }
+  ]
+
+  const fetchComponent = async (): Promise<void> => {
+    // In the current package structure, it is practically impossible to import anything from here
+    // without introducing cyclic dependencies, since text-editor-resources is in the dependency
+    // hierarchy of almost all other packages.
+    // In the future, it might be worth considering putting text editor plugins in the text-editor-plugins
+    // package so that one can create dependencies on packages such as activity and chunter.
+    const component = await getResource('chunter:component:InlineCommentThread' as AnyComponent)
```

Comment:
There should be no problem depending on `chunter` plugin as it does not depend on `text-editor-resources`